### DISCUSSION TITLE
Performance Suggestions? 23 minutes upscale 2048 x2 (-> 4096) on a 3060

### DISCUSSION THREAD

**Discussion by ghostsquad** [ssitu/ComfyUI_UltimateSDUpscale#52] [General]
I have a 3060 with 12GB of VRAM, and it takes 22.86 minutes (1372 seconds) to upscale from 2048x2048 to 4096x4096. What kinds of parameters can tweak to improve this performance?

![image](https://github.com/ssitu/ComfyUI_UltimateSDUpscale/assets/903488/d14cb953-2559-4fe6-8383-304044c0afe1)

**Comment by ssitu** [comment]
Are you able to increase the size of the tiles to 2048x2048? From a glance, it looks like you are generating 16 tiles, so you can reduce the number of tiles by increasing the size of the tiles. If that doesn't give good results, then you can reduce the steps, or maybe you're using conditioning that adds a lot of time to the sampling time. I may add a setting in the future for how many tiles you want to make at a time, like a batch.

**Reply by ghostsquad** [reply]
ya, I can do that. I did another test, going from 1024->2048 takes 358.23 seconds (same settings).

I'm adding controlnet openpose and depth conditioning.

**Reply by ghostsquad** [reply]
Running without controlnets, gives 312.35 seconds (upscaling from 1024->2048)

**Reply by ssitu** [reply]
You can also try reducing the tile_padding, since that increases the size of the tile, but you only have it at 32 so it probably won't make that much of a difference. Other than that, there's not much you can do relating to this node. There's nodes that approximate things to speed up sampling like tomesd, hypertile, and probably others.
